inst/include/dust/rng_interface.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef DUST_RNG_INTERFACE_HPP
+#define DUST_RNG_INTERFACE_HPP
+
+#include <R_ext/Random.h>
+
+template <typename T>
+std::vector<uint64_t> as_rng_seed(cpp11::sexp r_seed) {
+  auto seed_type = TYPEOF(r_seed);
+  std::vector<uint64_t> seed;
+  if (seed_type == INTSXP || seed_type == REALSXP) {
+    int seed_int = cpp11::as_cpp<int>(r_seed);
+    if (seed_int <= 0) {
+      cpp11::stop("'seed' must be non-negative");
```

Comment:
A negative seed might be used, I've certainly seen -1 used before. Can we not (quite safely for this purpose) cast the int as an unsigned int, and also thereby retain the full possible range [0-2^64) for the seed?

Reply:
sure - I think I was mixing up the requirement for nonzero state in xoshiro with the requirements of the intermediate generator here

Reply:
Done in 0b52ba3